### PR TITLE
New Tool - Quota Monitoring and Alerting 

### PR DESCRIPTION
**Pull request by anuradha-bajpai-google**
Quota Monitoring and Alerting tool allows GCP customers to view aggregated quotas across projects in an organization or folder. This tool also allows customers to configure threshold for quota usage and receive alerts.

### CODE REVIEW

**Comment by boredabdel** [comment]
Hi Anarudha, 

I can review this one for you. But first please have a look at the Cloud Build status, it failed for some reason https://pantheon.corp.google.com/cloud-build/builds/50f46d15-b5ba-41ad-b5e1-73fb5cc02355?project=cloud-eng-council

**Comment by boredabdel** [comment]
Checking of a comment re-trigger Cloud Build

**Comment by boredabdel** [comment]
@googlebot recheck CLA please

**Review by mike22664 — APPROVED** [review]

**Review by boredabdel — COMMENTED** [review]
Added few comments, please have a look

Also the Java files need formatting. Please download the java formatting tool 

https://github.com/google/google-java-format/releases/tag/google-java-format-1.7

and run it like this 

java -jar /usr/share/java/google-java-format-1.7-all-deps.jar -i 

against all the java files

File: tools/quota-monitoring-alerting/README.md [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+gcloud organizations add-iam-policy-binding  $TARGET_ORG_ID --member="serviceAccount:$SERVICE_ACCOUNT_ID@$DEFAULT_PROJECT_ID.iam.gserviceaccount.com"  --role="roles/compute.networkViewer" --condition=None
+
+gcloud organizations add-iam-policy-binding  $TARGET_ORG_ID --member="serviceAccount:$SERVICE_ACCOUNT_ID@$DEFAULT_PROJECT_ID.iam.gserviceaccount.com"  --role="roles/compute.viewer" --condition=None
+
+gcloud organizations add-iam-policy-binding  $TARGET_ORG_ID --member="serviceAccount:$SERVICE_ACCOUNT_ID@$DEFAULT_PROJECT_ID.iam.gserviceaccount.com"  --role="roles/resourcemanager.folderViewer" --condition=None
+
+gcloud organizations add-iam-policy-binding  $TARGET_ORG_ID --member="serviceAccount:$SERVICE_ACCOUNT_ID@$DEFAULT_PROJECT_ID.iam.gserviceaccount.com"  --role="roles/resourcemanager.organizationViewer" --condition=None
+```
+### 3.5 Download Service Account Key File
+1. Create Service Account key from host project A
```
[the remark this answers is not in the document]
Comment (anuradha-bajpai-google) [reply tools/quota-monitoring-alerting/README.md]: Thanks! Done!

File: tools/quota-monitoring-alerting/README.md [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+<img src="img/create_service_account_key.png" align="center" />
+2. Click on ‘Create’. This will download the key. Copy the file to a local directory and rename to CREDENTIALS_FILE.json
+<img src="img/download_service_account_key.png" align="center" />
+
+```
+mv *.json CREDENTIALS_FILE.json
+```
+### 3.6 Download Terraform File
+Run below command in the same directory as Service Account (in the previous step) to download the terraform file:
```
[the remark this answers is not in the document]
Comment (anuradha-bajpai-google) [reply tools/quota-monitoring-alerting/README.md]: To deploy the application, users can download only main.tf and they need to run it locally. They don't need to download the complete source code. They can add the link of the source code in the main.tf.

File: tools/quota-monitoring-alerting/main.tf [hunk 3]
```diff
@@ -0,0 +1,418 @@
+/*
```
[the remark this answers is not in the document]
Comment (anuradha-bajpai-google) [reply tools/quota-monitoring-alerting/main.tf:1]: Done!

File: tools/quota-monitoring-alerting/README.md [hunk 4]
```diff
@@ -0,0 +1,481 @@
...
+```
+gsutil cp gs://quota-monitoring-solution-source/main.tf .
+```
+
+
+4. Verify that you have these 2 files in your local directory:
+   - CREDENTIALS_FILE.json
+   - main.tf
+### 3.7 Configure Terraform
```
[the remark this answers is not in the document]
Comment (anuradha-bajpai-google) [reply tools/quota-monitoring-alerting/README.md:315]: Done. Moved the terraform files in terraform directory

File: tools/quota-monitoring-alerting/quota-notification/src/test/java/functions/SendNotificationTest.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
+package functions;
```
Comment (boredabdel) [inline tools/quota-monitoring-alerting/quota-notification/src/test/java/functions/SendNotificationTest.java:1]: This is a source code file and it requires the license header.

**Comment by boredabdel** [comment]
gcbrun

**Review by boredabdel — APPROVED** [review]
LGTM

**Pull request merged**